packages/content-management/table_list_view_table/src/components/use_tag_filter_panel.tsx

```diff
@@ -52,8 +55,26 @@ export const useTagFilterPanel = ({
   // "isInUse" state which disable the transition.
   const [isInUse, setIsInUse] = useState(false);
   const [options, setOptions] = useState<TagOptionItem[]>([]);
-  const [tagSelection, setTagSelection] = useState<TagSelection>({});
-  const totalActiveFilters = Object.keys(tagSelection).length;
+  const [tagSelection, setTagSelection] = useState<TagSelection>(
+    tagReferences
+      ? tagReferences.reduce((acc, obj) => {
+          acc[obj.name] = 'include';
+          return acc;
+        }, {} as TagSelection)
+      : {}
+  );
+  const totalActiveFilters = useMemo(
+    () =>
+      Object.keys(tagSelection).reduce((acc, currentOption) => {
+        const inTagReferences = tagReferences?.find((ref) => ref.name === currentOption);
+        acc +=
+          inTagReferences != null
+            ? tagReferences?.filter((ref) => ref.name === currentOption).length ?? 0
+            : 1;
+        return acc;
+      }, 0),
+    [tagReferences, tagSelection]
+  );
```

Comment:
This is to fix this scenario:
<img width="515" alt="incorrect_tag_counts" src="https://github.com/elastic/kibana/assets/6295984/2143b4b2-bb9a-41f7-8fd7-e91811504e75">